WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift

```diff
@@ -494,10 +495,41 @@ class ReaderDetailCoordinator {
     /// Given a URL presents it in a new Reader detail screen
     ///
     private func presentReaderDetail(_ url: URL) {
-        let readerDetail = ReaderDetailViewController.controllerWithPostURL(url)
+
+        // If the user tapped the link to the original post, use the original post's info to display reader detail.
+        // The API endpoint used by controllerWithPostID returns subscription flags for the post.
+        // The API endpoint used by controllerWithPostURL does return this information.
+        // These flags needed to display the `Follow conversation by email` option.
+        // So if we can call controllerWithPostID, do so. Otherwise, fallback to controllerWithPostURL.
+        // Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/17158
```

Comment:
Should we preface this comment with something making it clear it's regarding cross posts?
We might be missing a word or two in these comments. 
Should `The API endpoint used by controllerWithPostURL does return this information` read `does not`? 
Should `These flags needed to display the `Follow conversation by email` option` say `These flags are`?

Reply:
What? You got the gist right? 😉 Thank you, I'll fix it.